src/song_utils.py

```diff
@@ -8,12 +8,52 @@
 from mutagen.id3 import ID3FileType, PictureType
 from mutagen.ogg import OggFileType
 from mutagen.wave import WAVE
-from nextcord import File
+from nextcord import Attachment, Embed, File, Member, Message, User
+from nextcord.utils import escape_markdown
 from PIL import Image, UnidentifiedImageError
 
 import config
 
 
+def embed_song(
+    submit_message: Message | str,
+    attachment_filepath: str,
+    attachment: Attachment,
+    user: User | Member,
+    random_emoji,
+    jump_url=None,
+) -> Embed:
+    if isinstance(submit_message, Message) is False:
+        embed_title = f"{random_emoji} Now Previewing {random_emoji}"
+        message: str = submit_message
+        url = jump_url
+    else:
+        embed_title = f"{random_emoji} Now Playing {random_emoji}"
+        message: str = submit_message.content
+        url = submit_message.jump_url
+
+    # Build and send "Now Playing" embed
+    list_format = "{0}: [{1}]({2}) [`↲jump`]({3})"
+    embed_content = list_format.format(
+        user.mention,
+        escape_markdown(song_format(attachment_filepath, attachment.filename)),
+        attachment.url,
+        url,
+    )
+    embed = Embed(
+        title=embed_title, description=embed_content, color=config.PLAY_EMBED_COLOR
+    )
+
+    if message:
+        if len(message) > config.EMBED_FIELD_VALUE_LIMIT:
```

Comment:
I'm not sure why you unwrapped what was there before with a single if into an if-else, it causes code duplication in the last line. You should make it just an if and share the last line again

Reply:
I changed this due to the edge case with going between the Message object and a string for the `message_content`.

With your suggested changes, this can go back to how it was.